pkg/query-service/app/logs/v4/query_builder.go

```diff
@@ -29,3 +35,492 @@ const (
+		// with noop any filter or different order by other than ts will use new table
+		sqlSelect := constants.LogsSQLSelectV2
+		queryTmpl := sqlSelect + "from signoz_logs.%s where %s%s order by %s"
+		query := fmt.Sprintf(queryTmpl, DISTRIBUTED_LOGS_V2, timeFilter, filterSubQuery, orderBy)
+		return query, nil
+		// ---- NOOP ends here ----
+	}
+
+	// ---- FOR aggregation queries ----
+
+	// get the having conditions
+	having := logsV3.Having(mq.Having)
+	if having != "" {
+		having = " having " + having
+	}
+
+	// get the group by clause
+	groupBy := logsV3.GroupByAttributeKeyTags(panelType, graphLimitQtype, mq.GroupBy...)
+	if panelType != v3.PanelTypeList && groupBy != "" {
+		groupBy = " group by " + groupBy
+	}
+
+	// get the aggregation key
+	aggregationKey := ""
+	if mq.AggregateAttribute.Key != "" {
+		aggregationKey = getClickhouseKey(mq.AggregateAttribute)
+	}
+
+	// for limit queries, there are two queries formed
+	// in the second query we need to add the placeholder so that first query can be placed
+	if graphLimitQtype == constants.SecondQueryGraphLimit {
+		filterSubQuery = filterSubQuery + " AND " + fmt.Sprintf("(%s) GLOBAL IN (", logsV3.GetSelectKeys(mq.AggregateOperator, mq.GroupBy)) + "#LIMIT_PLACEHOLDER)"
+	}
+
+	aggClause, err := generateAggregateClause(mq.AggregateOperator, aggregationKey, step, preferRPM, timeFilter, filterSubQuery, groupBy, having, orderBy)
+	if err != nil {
+		return "", err
+	}
+
+	var queryTmplPrefix string
+	if graphLimitQtype == constants.FirstQueryGraphLimit {
+		queryTmplPrefix = "SELECT"
+	} else if panelType == v3.PanelTypeTable {
+		queryTmplPrefix =
+			"SELECT now() as ts,"
+		// step or aggregate interval is whole time period in case of table panel
+		step = (utils.GetEpochNanoSecs(end) - utils.GetEpochNanoSecs(start)) / NANOSECOND
+	} else if panelType == v3.PanelTypeGraph || panelType == v3.PanelTypeValue {
+		// Select the aggregate value for interval
+		queryTmplPrefix =
+			fmt.Sprintf("SELECT toStartOfInterval(fromUnixTimestamp64Nano(timestamp), INTERVAL %d SECOND) AS ts,", step)
+	}
+
+	query := queryTmplPrefix + selectLabels + aggClause
+
+	// for limit query this is the first query,
+	// we don't the the aggregation value here as we are just concerned with the names of group by
+	// for applying the limit
+	if graphLimitQtype == constants.FirstQueryGraphLimit {
+		query = "SELECT " + logsV3.GetSelectKeys(mq.AggregateOperator, mq.GroupBy) + " from (" + query + ")"
+	}
+	return query, nil
+}
+
+func buildLogsLiveTailQuery(mq *v3.BuilderQuery) (string, error) {
+	filterSubQuery, err := buildLogsTimeSeriesFilterQuery(mq.Filters, mq.GroupBy, v3.AttributeKey{})
+	if err != nil {
+		return "", err
+	}
+
+	switch mq.AggregateOperator {
+	case v3.AggregateOperatorNoOp:
+		query := constants.LogsSQLSelectV2 + "from signoz_logs." + DISTRIBUTED_LOGS_V2 + " where "
+		if len(filterSubQuery) > 0 {
+			query = query + filterSubQuery + " AND "
+		}
```

Comment:
Does live tail not use the resource filter?

Reply:
I have use the previous logic and update the code to support it. 

See commit https://github.com/SigNoz/signoz/pull/5908/commits/34c2c7b236db8a3b7e5cc8188b5419756877601d

This is how it will be used by the reader https://github.com/SigNoz/signoz/pull/5914/commits/fc08bd93d043ab021a834452420fd111040db0bf (different PR)